Marlin/src/lcd/extui/lib/mks_ui/draw_acceleration_settings.cpp

```diff
@@ -160,200 +160,59 @@ static void event_handler(lv_obj_t * obj, lv_event_t event) {
   if (disp_state_stack._disp_state[disp_state_stack._disp_index] != ACCELERATION_UI) {
     disp_state_stack._disp_index++;
     disp_state_stack._disp_state[disp_state_stack._disp_index] = ACCELERATION_UI;
   }
   disp_state = ACCELERATION_UI;
 
   scr = lv_obj_create(NULL, NULL);
-
   lv_obj_set_style(scr, &tft_style_scr);
   lv_scr_load(scr);
   lv_obj_clean(scr);
-
   (void)lv_label_create(scr, TITLE_XPOS, TITLE_YPOS, machine_menu.AccelerationConfTitle);
-
   lv_refr_now(lv_refr_get_disp_refreshing());
 
   if (uiCfg.para_ui_page != 1) {
+    ZERO(public_buf_l);
+    sprintf_P(public_buf_l, PSTR("%.1f"), planner.settings.acceleration);
+    lv_screen_menu_item_1_edit(scr, machine_menu.PrintAcceleration, PARA_UI_POS_X, PARA_UI_POS_Y, event_handler, ID_ACCE_PRINT, 0, public_buf_l);
 
-    (void)lv_label_create(scr, PARA_UI_POS_X, PARA_UI_POS_Y + 10, machine_menu.PrintAcceleration);
-
-    buttonPrintValue = lv_btn_create(scr, PARA_UI_VALUE_POS_X, PARA_UI_POS_Y + PARA_UI_VALUE_V, PARA_UI_VALUE_BTN_X_SIZE, PARA_UI_VALUE_BTN_Y_SIZE, event_handler, ID_ACCE_PRINT);
-    labelPrintValue = lv_label_create_empty(buttonPrintValue);
-
-    line1 = lv_line_create(scr, NULL);
-    lv_ex_line(line1, line_points[0]);
-
-    (void)lv_label_create(scr, PARA_UI_POS_X, PARA_UI_POS_Y * 2 + 10, machine_menu.RetractAcceleration);
-
-    buttonRetraValue = lv_btn_create(scr, PARA_UI_VALUE_POS_X, PARA_UI_POS_Y * 2 + PARA_UI_VALUE_V, PARA_UI_VALUE_BTN_X_SIZE, PARA_UI_VALUE_BTN_Y_SIZE, event_handler, ID_ACCE_RETRA);
-    labelRetraValue = lv_label_create_empty(buttonRetraValue);
-
-    line2 = lv_line_create(scr, NULL);
-    lv_ex_line(line2, line_points[1]);
-
-    (void)lv_label_create(scr, PARA_UI_POS_X, PARA_UI_POS_Y * 3 + 10, machine_menu.TravelAcceleration);
-
-    buttonTravelValue = lv_btn_create(scr, PARA_UI_VALUE_POS_X, PARA_UI_POS_Y * 3 + PARA_UI_VALUE_V, PARA_UI_VALUE_BTN_X_SIZE, PARA_UI_VALUE_BTN_Y_SIZE, event_handler, ID_ACCE_TRAVEL);
-    labelTravelValue = lv_label_create_empty(buttonTravelValue);
-
-    line3 = lv_line_create(scr, NULL);
-    lv_ex_line(line3, line_points[2]);
+    ZERO(public_buf_l);
+    sprintf_P(public_buf_l, PSTR("%.1f"), planner.settings.retract_acceleration);
+    lv_screen_menu_item_1_edit(scr, machine_menu.RetractAcceleration, PARA_UI_POS_X, PARA_UI_POS_Y * 2, event_handler, ID_ACCE_RETRA, 1, public_buf_l);
 
-    (void)lv_label_create(scr, PARA_UI_POS_X, PARA_UI_POS_Y * 4 + 10, machine_menu.X_Acceleration);
+    ZERO(public_buf_l);
+    sprintf_P(public_buf_l, PSTR("%.1f"), planner.settings.travel_acceleration);
+    lv_screen_menu_item_1_edit(scr, machine_menu.TravelAcceleration, PARA_UI_POS_X, PARA_UI_POS_Y * 3, event_handler, ID_ACCE_TRAVEL, 2, public_buf_l);
 
-    buttonXValue = lv_btn_create(scr, PARA_UI_VALUE_POS_X, PARA_UI_POS_Y * 4 + PARA_UI_VALUE_V, PARA_UI_VALUE_BTN_X_SIZE, PARA_UI_VALUE_BTN_Y_SIZE, event_handler, ID_ACCE_X);
-    labelXValue = lv_label_create_empty(buttonXValue);
+    ZERO(public_buf_l);
```

Comment:
I don't think these are needed in front of string population. The `sprintf_P` function is not likely to bail and return without writing to the buffer.

Reply:
True.. I'm too much focused in moving and rewriting those menus that something pass. I will remove all those zero.